rest/model/dns/record.go

```diff
@@ -75,3 +78,19 @@ func (r *Record) AddFilter(fil *filter.Filter) {
 
 	r.Filters = append(r.Filters, fil)
 }
+
+// Remove any dot (.) character prefix or suffix of the given string.
+// This is useful because dns names often end in '.' characters to signify
+// the root of the DNS tree.  But NS1 API does not support this.
+//
+// In other cases a domain or zone may be passed in with a preceding dot (.)
+// character which would likewise lead the system to fail.
+func removeEnclosingDots(inputString string) string {
+	if strings.HasSuffix(inputString, ".") {
+		inputString = inputString[0 : len(inputString)-1]
+	}
+	if strings.HasPrefix(inputString,"."){
+		inputString = inputString[1:len(inputString)]
+	}
+	return inputString
+}
```

Comment:
looks like these changes snuck in from the other PR. Want to think about the other one a bit, so if you wanna pull these out, we can get this stuff merged without waiting on it :)

Reply:
Good point.  This PR was just low-hanging fruit from when I was working through linter errors. 
It's being fixed.